lib/effects/combined_effect.dart

```diff
@@ -16,6 +16,10 @@ class CombinedEffect extends PositionComponentEffect {
     bool isAlternating = false,
     Function onComplete,
   }) : super(isInfinite, isAlternating, onComplete: onComplete) {
+    assert(
+      effects.every((effect) => effect.component == null),
+      "No effects can be added to components from the start",
```

Comment:
I am a bit unclear on what this message means...

Reply:
The component has a list of effects that is added to it, but here there is a list of effects added to the effect so if it is also added to a component both the component and the CombinedEffect (or SequenceEffect) will modify the effect and it will get very unexpected behaviour.

Reply:
maybe something like: each effect can only be added once? I assume this is the same problem of adding the same effect normally to two distinct components right. so that might be more clear

Reply:
also, nit: single quotes :p